### DISCUSSION TITLE
Persist Plugin - docs - value is reset after second reload

### DISCUSSION THREAD

**Discussion by dsturm** [alpinejs/alpine#1834] [7. Bugs (archived)] [answered]
Hej @calebporzio,

I was facing  a issue in the docs of the persist plugin:

After increment in the [docs page](https://alpinejs.dev/plugins/persist#magic-persist) and reload the value shows as set. But right away the local storage is set to `0`, so after an second reload it is shown as `0`.
Further incrementing in the [`$persist` section](https://alpinejs.dev/plugins/persist#magic-persist) would not set the value in the [`How does it work?` section](https://alpinejs.dev/plugins/persist#how-it-works). That second component there might be(?) the reason, why the local storage is overwritten?

![alpine-persist-docs-reset](https://user-images.githubusercontent.com/384815/126750901-b6da46ec-8edd-4d5a-b2be-0935780dc955.gif)

So, currently I don't know if it is a bug itself or an issue with the docs scenario.

**Comment by SimoTod** [comment, marked as answer]
The plugin seems to get confused because there are 2 components persisting the same variable in that page. I'm not sure what's the expected behavior here (i guess it's open to discussion), should the last one modified always set the value? 
Another thing I find odd is that only the first component seems to get populated from localsorage when refreshing.

**Reply by dsturm** [reply]
Yep, that's what I meant. The second component currently does not recognize the persistent data. 🤔

**Reply by SimoTod** [reply]
I can't try because I'm from my mobile phone but I believe this line https://github.com/alpinejs/alpine/blob/main/packages/persist/src/index.js#L13 should be `setter(initial)` otherwise once a component is loaded, it clears out the value.

**Reply by robdekort** [reply]
If this is regarded as an issue I think it would solve what I ran into here: https://github.com/alpinejs/alpine/discussions/1831 where the second component doesn't recognise the persisted data.

**Reply by robdekort** [reply]
> I can't try because I'm from my mobile phone but I believe this line https://github.com/alpinejs/alpine/blob/main/packages/persist/src/index.js#L13 should be `setter(initial)` otherwise once a component is loaded, it clears out the value.

I can confirm this change indeed fixes this issue.

**Reply by robdekort** [reply]
Thank you for your great pr @SimoTod.
